pkg/devicetemplates/ttscsv.go

```diff
@@ -177,6 +182,116 @@ var csvFieldSetters = map[string]csvFieldSetterFunc{
 		dst.SupportsJoin = true
 		return []string{"root_keys.nwk_key.key", "supports_join"}, nil
 	},
+	"rx1_delay": func(dst *ttnpb.EndDevice, val string) ([]string, error) {
+		var rx1Delay ttnpb.RxDelay
+		if err := rx1Delay.UnmarshalText([]byte(val)); err != nil {
+			return nil, err
+		}
+		if dst.MacSettings == nil {
+			dst.MacSettings = &ttnpb.MACSettings{}
+		}
+		dst.MacSettings.Rx1Delay = &ttnpb.RxDelayValue{
+			Value: rx1Delay,
+		}
+		return []string{"mac_settings.rx1_delay"}, nil
+	},
+	"supports_32_bit_f_cnt": func(dst *ttnpb.EndDevice, val string) ([]string, error) {
+		var supports32BitFCnt ttnpb.BoolValue
+		if err := supports32BitFCnt.UnmarshalText([]byte(val)); err != nil {
+			return nil, err
+		}
+		if dst.MacSettings == nil {
+			dst.MacSettings = &ttnpb.MACSettings{}
+		}
+		dst.MacSettings.Supports_32BitFCnt = &supports32BitFCnt
+		return []string{"mac_settings.supports_32_bit_f_cnt"}, nil
+	},
+	"dev_addr": func(dst *ttnpb.EndDevice, val string) ([]string, error) {
+		var devAddr types.DevAddr
+		if dst.Session == nil {
+			dst.Session = &ttnpb.Session{}
+		}
+		if err := devAddr.UnmarshalText([]byte(val)); err != nil {
+			return nil, err
+		}
+		dst.Session.DevAddr = devAddr[:]
+		return []string{"session.dev_addr"}, nil
+	},
+	"app_s_key": func(dst *ttnpb.EndDevice, val string) ([]string, error) {
+		key := &types.AES128Key{}
+		if err := key.UnmarshalText([]byte(val)); err != nil {
+			return nil, err
+		}
+		if dst.Session == nil {
+			dst.Session = &ttnpb.Session{}
+		}
+		if dst.Session.Keys == nil {
+			dst.Session.Keys = &ttnpb.SessionKeys{}
+		}
+		if dst.Session.Keys.AppSKey == nil {
+			dst.Session.Keys.AppSKey = &ttnpb.KeyEnvelope{}
+		}
```

Comment:
I don't think this field gets set outside this function. You can skip this and directly assume that `AppSKey` is nil